homework/index.js

```diff
@@ -30,13 +30,75 @@
     return elem;
   }
 
+  function pageConstructor(data) {
+    data.sort((a, b) => a.name.localeCompare(b.name));
+    const root = document.getElementById('root');
+    const headerDiv = createAndAppend('div', root, { class: 'headerDiv' });
+    createAndAppend('p', headerDiv, { text: 'HYF Repositories' });
+    const headerSelect = createAndAppend('select', headerDiv, { class: 'headerSelect' });
+    data.map((elem, index) =>
+      createAndAppend('option', headerSelect, { text: elem.name, value: index }),
+    );
+    const container = createAndAppend('div', root, { class: 'container' });
+    const repoInfoDiv = createAndAppend('div', container, { class: 'repoInfoDiv' });
+    const ul = createAndAppend('ul', repoInfoDiv);
+    const repository = createAndAppend('li', ul, { text: `Repository :` });
+    createAndAppend('a', repository, {
+      text: `${data[0].name}`,
+      href: data[0].html_url,
+      target: '_blank',
+      id: 'repository',
+    });
+    createAndAppend('li', ul, { text: `Description :${data[0].description}`, id: 'description' });
+    createAndAppend('li', ul, { text: `Forks :${data[0].forks}`, id: 'forks' });
+    createAndAppend('li', ul, { text: `Update :${data[0].updated_at}`, id: 'update' });
+    const contributorDiv = createAndAppend('div', container, { class: 'contributorDiv' });
+    createAndAppend('p', contributorDiv, { text: 'Contributions' });
+    const contributorUl = createAndAppend('ul', contributorDiv, { class: 'contributorUl' });
+    function contributor() {
```

Comment:
I see this function just in-collapses it's content to be executed the very next line to it
But it became handy on line [88] when we wish to have it called in the event by it self

If I where you I would recommend moving this function to the file level then having some arguments to passing it's mainly used `variables/values/etc...` as a parameters and that is the main reason of functions 
> declering somewhere and calling in other place

making code look shorter for a human read and making `pageConstructor` function less 18 lines long then it currently is, and `contributor` is a small function so what to say about functions around a 100 lines long [which you may see], and that will make the parent function a huge one for one or two things..